### PR TITLE
Add toasts to modals

### PR DESCRIPTION
**Pull request by janKozie1**
for: https://app.zenhub.com/workspaces/openstax-unified-5b71aabe3815ff014b102258/issues/openstax/unified/1298

### CODE REVIEW

**Review by PiotrKozlowski — CHANGES_REQUESTED** [review]
Most of my comments are minor
this one I think is important: 

One of the AC for error bar is:
> remains floating when page is scrolled

I think this is because when you loaded first page of highlights and then scroll down to scroll another and ther is an error - you can't see the bar

**Review by PiotrKozlowski — COMMENTED** [review]
When I've disabled internet and scrolled to the bottom of the modal I can see that multiple requests are dispatched: https://drive.google.com/file/d/16TNelCJXh7onGeUSbwnDwX0N60ow9ysT/view?usp=sharing 

If this is something easy to fix I think it would be good to fix in this pr and if it doesn't then just let Malar know so she can decide what to do with it

**Comment by janKozie1** [comment]
I think it would be best to implement an intersection obeserver instead of the onScroll events. I will ask Malar if she wants it fixed in this card.

File: src/app/notifications/utils.ts [hunk 1]
```diff
@@ -16,3 +19,15 @@ export const shouldLoadAppMessage = (message: Message) => {
 
   return true;
 };
+
+export const getHighlightToastDesination = (state: AppState): ToastNotification['destination'] =>
+  myHighlightsOpen(state) ? 'myHighlights' : 'page';
+
+export const groupToasts = (toasts: ToastNotification[]) => toasts.reduce((grouped, toast) => {
```
[the remark this answers is not in the document]
Comment (janKozie1) [reply src/app/notifications/utils.ts:26]: I was avoiding hard coding a list of possible destinations. This way only type needs to be updated when adding a new destination.

**Review by PiotrKozlowski — APPROVED** [review]

**Pull request merged**